DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -228,34 +228,15 @@ public void PromoteDelegateToAdmin(AdminRoles adminRoles, int? categoryId, int d
         {
             var delegateUser = userDataService.GetDelegateUserById(delegateId)!;
 
-            if (string.IsNullOrWhiteSpace(delegateUser.EmailAddress) ||
-                string.IsNullOrWhiteSpace(delegateUser.FirstName) ||
-                string.IsNullOrWhiteSpace(delegateUser.Password))
-            {
-                throw new AdminCreationFailedException(
-                    "Delegate missing first name, email or password",
-                    AdminCreationError.UnexpectedError
-                );
-            }
-
-            var adminUser = userDataService.GetAdminUserByEmailAddress(delegateUser.EmailAddress);
+            var userId = userDataService.GetUserIdFromDelegateId(delegateId);

```

Comment:
There is a new GetDelegateEntityById coming in HEEDLS-951. If that gets merged first, it might be worth using that instead of this pair of calls.